servicepulse/intro-failed-messages.md

```diff
@@ -1,72 +1,93 @@
+
+The display of failed message groups can be changed via the "Group by" drop down menu, according to the following classifications types:
+
+ * **Exception Type and Stack Trace** - groups messages both by exception type and stack trace. It is the default way of categorizing failed messages.   
+ * **Message Type** - groups messages by message type. 
+ 
+Note: the number of listed groups may differ depending on the selected classifications type view.
 
+###### Managing failed message groups
 
-### Failed Message Retry
+Regardless of classification type, you can take the following actions on failed message groups:
 
-After addressing the root cause of the message's processing failure, resend the failed message for reprocessing by the endpoint(s). This is referred to as a "retry" (or a manual retry, in contrast to the automated and configurable [Recoverability](/nservicebus/recoverability/)).
+ * **View messages** - Shows all individual messages contained in the group.
+ * **Request retry** - Sends all failed messages to the corresponding queue to attempt processing again. When a failed group retry request is initiated, ServicePulse will present the progress of the operation.
 
-To retry a failed message, select the failed message(s) in the failed messages list and click the "Retry Selected" button (or click "Retry Group").
+![Failed message groups retry in progress](images/failed-group-retry-in-progress.png 'width=500')
 
-A message that is sent for retry is marked as such, and is not displayed in the failed message list or included in failed message groups, unless the reprocessing fails again. Messages sent for retry appear in the [Pending Retries screen](intro-pending-retries.md) until either an audit message is received indicating success, the message is returned as an error, or the message is manually marked as resolved.
+ * **Archive group** - Archives all messages contained in the group. [Learn more about archiving messages](/servicepulse/intro-archived-messages).
 
-If a message fails repeated retry attempts, an indication is added, including the number of times it has failed.
 
-![Repeated failure indication](images/failed-messages-repeated-failure.png 'width=500')
+#### All Messages tab and Failed Message group detail view
 
-NOTE: The number of retry attempts for a message can be significant if the handler for that message is not [idempotent](/nservicebus/concept-overview.md#idempotence). Any processing attempt that invokes logic that does not participate in the NServiceBus transactional processing will not be rolled back on processing failure.
+The "View messages" link in a failed message group entry and the "All messages" tab display the individual failed messages inside that failed group or all individual failed messages, respectively.
 
+Note: Working with individual messages can be useful for testing system fixes before deciding to do any potentially time consuming bulk operations.
```

Comment:
> Working with individual messages

Not sure what that means, better "It's useful to test the fix by retrying a single message first, and request retrying the whole group only if the individual retry/operation was successful."